### PR TITLE
mod : updated item value model

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

File: src/Module.Server/Common/Models/CrpgItemValueModel.cs [hunk 1]
```diff
@@ -86,20 +88,24 @@ private float CalculateArmorTier(ArmorComponent armorComponent)
...
-            ItemObject.ItemTypeEnum.HorseHarness => 10 * armorPower / bestArmorPower,
-            _ => 10 * armorPower / (bestArmorPower * (float)Math.Pow(armorComponent.Item.Weight + 8, 0.5f)),
+
+            ItemObject.ItemTypeEnum.HorseHarness => 10 * armorPower / bestArmorPower / (float)Math.Pow(1 + heirloomLevel / 10f, 1f),
+            _ => (10 * armorPower) / (bestArmorPower * (float)Math.Pow(armorComponent.Item.Weight + 8, 0.5f)) / (float)Math.Pow(1 + heirloomLevel / 20f, 1f),
         };
     }
 
-    private float CalculateHorseTier(HorseComponent horseComponent)
+    private float CalculateHorseTier(ItemObject item, HorseComponent horseComponent)
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgItemValueModel.cs:97]: You can the item with `horseComponent.Item` like in the other methods

File: src/Module.Server/Common/Models/CrpgItemValueModel.cs [hunk 2]
```diff
@@ -397,4 +419,28 @@ private float CalculateAmmoTier(WeaponComponent weaponComponent)
...
+            else
+            {
+                return 0;
+            }
+        }
+        else
+        {
+            return 0;
+        }
+    }
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgItemValueModel.cs:445]:
```suggestion
        if (!item.StringId.StartsWith("crpg_"))
        {
            return 0;
        }
        
        if (!int.TryParse(item.StringId.Split('_').Last().Substring(1), out int heirloomLevel))
        {
            return 0;
        }
        
        return heirloomLevel;
    }

```

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Module.Server/Common/Models/CrpgItemValueModel.cs (hunk 1 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgItemValueModel.cs:97]: no i can't it throws a null exception

Comment (verdie-g) [reply src/Module.Server/Common/Models/CrpgItemValueModel.cs:97]: Classic

**Review by verdie-g — APPROVED** [review]

**Pull request merged**